service/routes.go

```diff
@@ -16,45 +16,45 @@ import (
 )
 
 func (svc *Service) buildRoutes(ctx context.Context) {
-	if svc.config.EnablePrivateEndpoints {
-		svc.router = svc.privateEndpoints(ctx)
+	svc.router = chi.NewRouter()
+
+	svc.router.Handle("/health", http.HandlerFunc(svc.HealthCheck.Handler))
+
+	if svc.Config.EnablePrivateEndpoints {
+		svc.privateEndpoints(ctx)
 	} else {
-		svc.router = svc.publicEndpoints(ctx)
+		svc.publicEndpoints(ctx)
 	}
-
-	svc.router.Handle("/health", http.HandlerFunc(svc.healthCheck.Handler))
 }
 
-func (svc *Service) publicEndpoints(ctx context.Context) *chi.Mux {
+func (svc *Service) publicEndpoints(ctx context.Context) {
 	log.Info(ctx, "enabling public endpoints")
 
-	r := chi.NewRouter()
-
-	hello := handler.NewHello(svc.responder, svc.cantabularClient)
-	r.Get("/hello", hello.Get)
-	r.Post("/hello", hello.Create)
+	// Routes
+	areaTypes := handler.NewAreaTypes(svc.responder, svc.cantabularClient)
 
-	return r
+	svc.router.Get("/area-types", areaTypes.Get)
 }
 
-func (svc *Service) privateEndpoints(ctx context.Context) *chi.Mux {
+func (svc *Service) privateEndpoints(ctx context.Context) {
 	log.Info(ctx, "enabling private endpoints")
 
+	// use sub-router to prevent /health endpoint from requiring auth
 	r := chi.NewRouter()
 
 	// Middleware
-	svc.identityClient = identity.New(svc.config.ZebedeeURL)
+	svc.identityClient = identity.New(svc.Config.ZebedeeURL)
 	checkIdentity := dphandlers.IdentityWithHTTPClient(svc.identityClient)
-	permissions := middleware.NewPermissions(svc.config.ZebedeeURL, svc.config.EnablePermissionsAuth)
+	permissions := middleware.NewPermissions(svc.Config.ZebedeeURL, svc.Config.EnablePermissionsAuth)
 
 	r.Use(permissions.Require(auth.Permissions{Read: true}))
 	r.Use(checkIdentity)
 	r.Use(middleware.LogIdentity())
 
 	// Routes
-	hello := handler.NewHello(svc.responder, svc.cantabularClient)
-	r.Get("/hello", hello.Get)
-	r.Post("/hello", permissions.RequireCreate(hello.Create))
+	areaTypes := handler.NewAreaTypes(svc.responder, svc.cantabularClient)
+
+	r.Get("/area-types", areaTypes.Get)
```

Comment:
wouldn't it be better to do this and line 36 in one place to avoid repeating route definitions?

Reply:
I think we want to keep the oportunity to define different endpoints dependinig on the 'private' or 'public' flag state.
For example, in dataset api, the POST and PUT endpoints are not available publicly. 

In this case, the GET is available in both cases, but i think we should keep the duplication to allow for new endpoints to be defined in one case only if required.